x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -131,98 +131,107 @@ private void createFollowerIndex(
-
-                // Adding the leader index uuid for each shard as custom metadata:
-                Map<String, String> metadata = new HashMap<>();
-                metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_SHARD_HISTORY_UUIDS, String.join(",", historyUUIDs));
-                metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_UUID_KEY, leaderIndexMetaData.getIndexUUID());
-                metadata.put(Ccr.CCR_CUSTOM_METADATA_LEADER_INDEX_NAME_KEY, leaderIndexMetaData.getIndex().getName());
-                metadata.put(Ccr.CCR_CUSTOM_METADATA_REMOTE_CLUSTER_NAME_KEY, request.getRemoteCluster());
-                imdBuilder.putCustom(Ccr.CCR_CUSTOM_METADATA_KEY, metadata);
-
-                // Copy all settings, but overwrite a few settings.
-                Settings.Builder settingsBuilder = Settings.builder();
-                settingsBuilder.put(leaderIndexMetaData.getSettings());
-                // Overwriting UUID here, because otherwise we can't follow indices in the same cluster
-                settingsBuilder.put(IndexMetaData.SETTING_INDEX_UUID, UUIDs.randomBase64UUID());
-                settingsBuilder.put(IndexMetaData.SETTING_INDEX_PROVIDED_NAME, followIndex);
-                settingsBuilder.put(CcrSettings.CCR_FOLLOWING_INDEX_SETTING.getKey(), true);
-                settingsBuilder.put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true);
-                imdBuilder.settings(settingsBuilder);
-
-                // Copy mappings from leader IMD to follow IMD
-                for (ObjectObjectCursor<String, MappingMetaData> cursor : leaderIndexMetaData.getMappings()) {
-                    imdBuilder.putMapping(cursor.value);
-                }
-                imdBuilder.setRoutingNumShards(leaderIndexMetaData.getRoutingNumShards());
-                IndexMetaData followIMD = imdBuilder.build();
-                mdBuilder.put(followIMD, false);
+                    @Override
+                    public void onResponse(RestoreService.RestoreCompletionResponse response) {
+                        afterRestoreStarted(clientWithHeaders, request, listener, response);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        listener.onFailure(e);
+                    }
+                });
+            }
+        });
+    }
 
-                ClusterState.Builder builder = ClusterState.builder(currentState);
-                builder.metaData(mdBuilder.build());
-                ClusterState updatedState = builder.build();
+    private void afterRestoreStarted(Client clientWithHeaders, PutFollowAction.Request request,
+                                     ActionListener<PutFollowAction.Response> originalListener,
+                                     RestoreService.RestoreCompletionResponse response) {
+        final ActionListener<PutFollowAction.Response> listener;
+        if (request.waitForActiveShards() == ActiveShardCount.NONE) {
```

Comment:
let's use `.equals()` instead of `==` (even though it's not strictly needed, it's less bugprone if we were to refactor this in the future)